src/VisualStudio/Xaml/Impl/Diagnostics/Analyzers/XamlDocumentDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Xaml;
+using Microsoft.VisualStudio.ComponentModelHost;
+using Package = Microsoft.VisualStudio.Shell.Package;
+
+namespace Microsoft.CodeAnalysis.Xaml.Diagnostics.Analyzers
+{
+    [DiagnosticAnalyzer(StringConstants.XamlLanguageName)]
+    internal class XamlDocumentDiagnosticAnalyzer : DocumentDiagnosticAnalyzer
+    {
+        private IXamlDocumentAnalyzerService analyzerService;
+
+        private IXamlDocumentAnalyzerService AnalyzerService
```

Comment:
WARNING!!!

We'll call your `Analyze*` methods on arbitrary threadpool threads.  Having to access global VS services delayed like this frightens me a lot.

I also worry about potential deadlocks if someone very quickly does something that blocks for diagnostics to be computed (like moving the caret and hitting `Ctrl+.` very quickly.